tools/commentMonitor/main.go

```diff
@@ -62,104 +101,144 @@ func main() {
 
 		owner := *e.GetRepo().Owner.Login
 		repo := *e.GetRepo().Name
-		prnumber := *e.GetIssue().Number
-
-		// Github client for posting comments.
-		clt := newClient(os.Getenv("GITHUB_TOKEN"))
+		pr := *e.GetIssue().Number
+		author := *e.Sender.Login
+		authorAssociation := *e.GetComment().AuthorAssociation
+		commentBody := *e.GetComment().Body
+
+		// Setup commentMonitorClient.
+		ctx := context.Background()
+		cmClient.ghClient = newGithubClient(ctx, owner, repo, pr)
+
+		// Validate comment if regexString provided.
+		if cmClient.regexString != "" {
+			cmClient.regex = regexp.MustCompile(cmClient.regexString)
+			if !cmClient.regex.MatchString(commentBody) {
+				log.Fatalf("matching command not found. comment validation failed")
+			}
+			log.Println("comment validation successful")
+		}
 
-		// Check author association.
-		if *verifyUser {
-			if err := memberValidation(*e.GetComment().AuthorAssociation); err != nil {
+		// Verify if user is allowed to perform activity.
+		if !cmClient.verifyUserDisabled {
+			var allowed bool
+			allowedAssociations := []string{"COLLABORATOR", "MEMBER", "OWNER"}
+			for _, a := range allowedAssociations {
+				if a == authorAssociation {
+					allowed = true
+				}
+			}
+			if !allowed {
 				log.Printf("author is not a member or collaborator")
-				if err := postComment(clt, owner, repo, prnumber, fmt.Sprintf("Error: %s is not a group member nor a collaborator and cannot execute benchmarks.", *e.Sender.Login)); err != nil {
-					log.Fatalln(err)
+				b := fmt.Sprintf("@%s is not a org member nor a collaborator and cannot execute benchmarks.", author)
+				if err := cmClient.ghClient.postComment(ctx, b); err != nil {
+					log.Fatalf("%v : couldn't post comment", err)
 				}
-				os.Exit(78)
+				os.Exit(1)
 			}
-			log.Printf("author is member or collaborator")
+			log.Println("author is a member or collaborator")
 		}
 
-		// Validate comment.
-		args, err := regexValidation(*regex, *e.GetComment().Body)
-		if err != nil {
-			log.Printf("matching command not found")
-			os.Exit(78)
-		}
+		// Extract args if regexString provided.
+		if cmClient.regexString != "" {
+			// To have unnamed arguments start at ARG_0 and so forth.
+			unnamedArgCount := 0
+
+			// Add comment arguments.
+			commentArgs := cmClient.regex.FindStringSubmatch(commentBody)[1:]
+			commentArgsNames := cmClient.regex.SubexpNames()[1:]
+			for i, argName := range commentArgsNames {
+				if argName == "" {
+					argName = fmt.Sprintf("ARG_%v", unnamedArgCount)
```

Comment:
as I mentioned in on of my comments I think it is best to remove the autogenerated names all together and enforce the use of named regex groups. It will simplify the code and avoid human errors for not using the correct arg file.

Reply:
oh alright, thanks I misunderstood the first comment. I'll add this.